mne/io/ant/tests/test_ant.py

```diff
@@ -0,0 +1,128 @@
+# Authors: The MNE-Python contributors.
+# License: BSD-3-Clause
+# Copyright the MNE-Python contributors.
+
+from __future__ import annotations
+
+import warnings
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import pytest
+from numpy.testing import assert_allclose
+
+from mne import Annotations
+from mne.datasets import testing
+from mne.io import BaseRaw, read_raw_ant, read_raw_brainvision
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+
+def read_raw_bv(fname: Path) -> BaseRaw:
+    """Read a brainvision file exported from eego."""
+    with warnings.catch_warnings():
```

Comment:
Can you add a comment about why some are out of range? Bug in eego? Bug/limitation with how we handle the annotations?

Reply:
Done, had to dig in a bit.. turns out the first impedance measurement is annotated 1 sample before the start of the recording. It still appears as an annotation, `raw.set_annotation` is smart enough to crop it instead of dropping it entirely.

Reply:
Ahh, in that case then instead of having warnings every time a file with impedances is read, it seems better to detect the impedance annotation's existence and adjust it/then all later by one sample.

Reply:
... But you can always do that later if you want

Reply:
Reading the full diff I now see you have some nice comment above so I think no need to change this 👍